.cbmc-batch/jobs/aws_cryptosdk_md_update/aws_cryptosdk_md_update_harness.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cryptosdk/private/cipher.h>
+#include <cbmc_invariants.h>
+#include <evp_utils.h>
+#include <make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+
+#include <cipher_openssl.h>
+
+void aws_cryptosdk_md_update_harness() {
+    /* arguments */
+    struct aws_cryptosdk_md_context md_context;  // can't be NULL according to preconditions
```

Comment:
In a previous comment you assume the structure could be NULL and right after you explicitly assume is not NULL, which I imagine it was for clarity/style reasons. Either you should update the previous proof, or keep the style on this one. In this proof, you do something similar for `buf`.